sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -711,7 +715,7 @@ nonReserved
     | ADD
     | OVER | PARTITION | RANGE | ROWS | PRECEDING | FOLLOWING | CURRENT | ROW | LAST | FIRST | AFTER
     | MAP | ARRAY | STRUCT
-    | LATERAL | WINDOW | REDUCE | TRANSFORM | USING | SERDE | SERDEPROPERTIES | RECORDREADER
```

Comment:
First, based on the link http://developer.mimer.se/validator/sql-reserved-words.tml, `USING` is a reserved word in SQL standard since SQL-92. 

Second, since 1.2, Hive introduces a flag `hive.support.sql11.reserved.keywords` for backward compatbility, which defaults to `true`. 
> Added In: Hive 1.2.0 with HIVE-6617: https://issues.apache.org/jira/browse/HIVE-6617
> Whether to enable support for SQL2011 reserved keywords. When enabled, will support (part of) SQL2011 reserved keywords.

In 2.2, Hive removes this flag and does not allow users to change it to false. That means, users are unable to use these reserved words as identifiers anymore, unless using them as quoted identifiers. See: https://issues.apache.org/jira/browse/HIVE-14872

Thus, I think it is safe to remove `USING` from the non-reserved words. 

cc @hvanhovell